src/package-linker.js

```diff
@@ -244,31 +244,39 @@ export default class PackageLinker {
       }
     }
 
-    // keep track of all scoped paths to remove empty scopes after copy
-    const scopedPaths = new Set();
-
-    // register root & scoped packages as being possibly extraneous
     const possibleExtraneous: Set<string> = new Set();
-    for (const folder of this.config.registryFolders) {
-      const loc = path.join(this.config.cwd, folder);
-
-      if (await fs.exists(loc)) {
-        const files = await fs.readdir(loc);
-        let filepath;
-        for (const file of files) {
-          filepath = path.join(loc, file);
-          if (file[0] === '@') {
+    const scopedPaths: Set<string> = new Set();
+
+    const findExtraneousFiles = async basePath => {
```

Comment:
It may be better in the future to extract this method outside of the closure.

Reply:
Since it needs to changes `scopedPaths` and `extraneousFiles` and isn't meant to be reused anywhere else in the code I thought it was cleaner to keep it together with the code segment that uses it.